playwright/tests/request-blocking.spec.ts

```diff
@@ -0,0 +1,22 @@
+import * as selectors from '../data/selectors.json';
+import { expect, test } from '@playwright/test';
+
+test.describe('Request Blocking', () => {
+    let beforeActionVisibility;
+    let afterActionVisibility;
+
+    test.beforeEach(async ({page}) => {
```

Comment:
I think you do not need to pump everything into the hook function. Moreover, `test.beforeEach` has nothing to do here. You can simply set up two test cases within this `describe`.

e.g.

- TC1 To verify the initial state of the page
- TC2 To verify the request

The benefit is that you can use the built-in visibility-specific [assertions](https://playwright.dev/docs/test-assertions#locator-assertions-to-be-visible) without using additional variables